### DISCUSSION TITLE
Non accedo da Internet tramite app android

### DISCUSSION THREAD

**Discussion by domoZAM** [indomus/forum#2354] [HUB: Home Assistant]
Buongiorno a tutti, ho un problema: dalla app di android riesco ad accedere al mio hub Homeassistant se mi trovo nella mia rete LAN di casa, viceversa, se sono fuori casa, e voglio sfruttare Internet l'app non si connette: continua a ritentare la connessione ma non arriva a completamento.
Se la stessa cosa la faccio dal browser sul telefonino (via Chrome), l'accesso invece funziona regolarmente. 
La mia App è in versione 2023.1.1-full. La raggiugibilità dell'hub di casa avviene tramite https e nome logico risolto via duckdns. Non è configurata un url dedicata per la connesione locale
L'hub di casa è in versione 2022.7.0.
Qualcuno può suggerirmi qualche modo per risolvere la cosa.. Lo stesso problema lo riscontro anche sui cellulari dei miei familiari il che mi porta ad escludere  l'idea di un app corrotta e della necessità di reinstallarla, che ne dite?
Grazie per ogni contributo.

**Comment by snakuzzo** [comment]
Dal comportamento che hai descritto, mi verrebbe da pensare ad una scadenza del certificato che hai esposto.
Da browser puoi tranquillamente bypassare il warning sul certificato non valido, e quindi andare avanti
Con la companion app invece no. Per collegarti devi avere per forza di cose un certificato valido.
Controlla questa cosa.

L'app "corrotta" è un'ipotesi assurda e da scartare

**Reply by domoZAM** [reply]
intanto grazie per la solerte attenzione, come faccio a sapere se il certificato "è valido" ?
io ho integrato nell'hub la verifica del certificato e questi mi riporta l'indicazione di  certificato in scadenza il 
2 Maggio 2023.. come scadenza quindi è valido..

**Reply by snakuzzo** [reply]
allora il problema non è quello.
Che problemi hai sull'app ?

**Reply by domoZAM** [reply]
Dall'app apparentemente non ho problemi: se la utilizzo in casa (quindi con cell. connesso in wifi alla rete locale che è la medesima dell'hub) si connette e posso tranquillamente "navigare" sui vari  folder ecc..
Se cerco di usarla quando sono in giro, oppure scollegando il cell. dal wifi di casa, esce la videata di connessione e poco dopo il messaggio di "Impossibile connettersi a Home assistant" con le tre voci "impostazioni", "aggiorna url esterno", "attendi"  ..da qui non si schioda nonostante ripetuti tentativi.
L'app è configurata per puntare l'url/porta che sono gli stessi impostati nell'integrazione per la verifica del certificato.

Non so se può essere significativo, ho notato (quando riesco ad utilizzare l'app), che andando nella lista dei token di aggiornamento (voce Profilo del menu) non vi sono token con url pubblica ma solo url con ip privati (della lan), però non ho ben capito cosa servano e come si aggiornano..

Grazie

**Reply by snakuzzo** [reply]
Non li vedi perché se non riesci a collegarti, non te li genera.
Mi sembra strano. Sicuro che nelle impostazioni dell'app tu abbia solo la URL esterna configurata ?
Mi fai vedere uno screenshot di come hai impostato quella sezione ?

**Reply by domoZAM** [reply]
**_Immagine rimossa_**

**Reply by snakuzzo** [reply]
Rimuovo l'immagine perché non avevi oscurato il tuo dominio.
Comunque tu stai usando la URL `https://tuodominio:5026`... a questo punto mi viene da pensare che sotto APN del tuo gestore mobile, non puoi raggiungere il sito sulla 5026.
Se sei in grado, ti basta fare un test con una qualsiasi app con utility di network e provare una telnet su tuodominio tuaporta.
E comunque ti consiglierei di provare ad esporre il tuo dominio su una porta diversa (tanto per dirne una, prova la 9000)

**Reply by domoZAM** [reply]
..però se lo punto da browser il sito mi risponde, mi propone login e password di HA

**Reply by snakuzzo** [reply]
mi fai vedere la URL come viene mostrata sul browser dello smartphone ?

**Comment by domoZAM** [comment]
![InkedScreenshot_2023-02-21-16-16-51_LI](https://user-images.githubusercontent.com/76051127/220385620-b3f5ba6e-2d8b-49cf-9c14-278126f204ba.jpg)

**Reply by snakuzzo** [reply]
Il certificato è ok.
Invece la schermata dell'app ?

**Comment by domoZAM** [comment]
La schermata dell'app quando funziona (in LAN) è la vista di caduta dell'hub. se mi disconnetto dall'app (Disconnessione) mi riparte questa videata:
![Screenshot_20230221-163207](https://user-images.githubusercontent.com/76051127/220390176-7656608c-fdbc-4c60-b072-ca3426a3321a.png)
seguita da questa:
![Screenshot_20230221-163301](https://user-images.githubusercontent.com/76051127/220390284-15ab61a4-d102-435e-be4c-e5e8b17b24dc.png)
 a cui fa seguito la ricerca del server.. qui impiega un bel po (sembrerebbe che non lo trovi) ma siccome c'è l'opzione "inserisci manualmente l'indirizzo" solitamente lo digito: https://miodom.duckdns.org:5026 a cui fa seguito la videata con la richiesta delle credenziali:
![Screenshot_20230221-163448](https://user-images.githubusercontent.com/76051127/220391088-273a96ad-3243-4583-a822-a3376038dfbd.png)

dopodichè si cade nella pagina principale dell'hub

**Reply by domoZAM** [reply]
invece da internet l'app rimane su queste due videate:
![Screenshot_20230221-164501](https://user-images.githubusercontent.com/76051127/220392791-884a5c45-53c7-483a-96e2-dd4e3f7ef9e9.png)
![Screenshot_20230221-164537](https://user-images.githubusercontent.com/76051127/220392950-740ba044-41c4-4c2a-ab82-3bfc5cc2a9f8.png)

**Reply by snakuzzo** [reply]
Per il discorso "da internet" non raggiunge la URL, c'è poco da dire.
Fai un test di raggiungibilità con una app apposita sul cellulare

**Reply by domoZAM** [reply]
Ok, grazie per le attenzioni, proverò..
Resta il fatto che con un browser la connessione si realizza..  evidentemente l'app ha qualche mimica di connessione differente..
Proverò  ad indagare, magari smontando la cifratura https,  vedendo se ottengo i medesimi risultati...
Grazie ancora, ti  terrò aggiornato.
